include/boost/math/tools/polynomial.hpp

```diff
@@ -455,6 +454,18 @@ class polynomial :
        normalize();
        return *this;
    }
+   
+   // Useful for fast test of equality to zero.
+   operator bool() const
```

Comment:
Ahh, wow, that's a new one on me. What about naming a function that sets the polynomial to zero? I have used the verb `clear` but maybe `set_zero` is more appropriate? This is on the assumption that it is more efficient than assignment from zero, which I assume the compiler is not quite smart enough to optimize away for a class such as polynomial.